flow/layers/layer.h

```diff
@@ -313,11 +319,25 @@ class Layer {
     AutoSaveLayer(const PaintContext& paint_context,
                   const SkCanvas::SaveLayerRec& layer_rec,
                   SaveMode save_mode = SaveMode::kInternalNodesCanvas);
+    AutoSaveLayer(DisplayListBuilder* builder,
+                  const SkRect& bounds,
+                  const DlPaint* paint,
+                  const DlImageFilter* backdrop = nullptr);
 
-    const PaintContext& paint_context_;
     const SkRect bounds_;
-    // The canvas that this layer is saved on and popped from.
-    SkCanvas& canvas_;
+    enum class Type { kSkia, kDisplayList };
+    Type type_;
+    union {
```

Comment:
I would rather see subclasses - AutoSaveLayer -> (SkAutoSaveLayer, DlAutoSaveLayer)

Also, don't both need paint_context to manage the checkerboarding in the destructor?